### ISSUE TITLE
In-app message not appearing consistently

### ISSUE THREAD

**Issue by marcgug** [Appboy/appboy-ios-sdk#118]
* [X] I have checked [Appboy's documentation](https://www.appboy.com/documentation/iOS/) and am still experiencing the issue.

# Report
When previewing or running certain in-app campaigns, the in-app message only appears once per user. Any time it it triggered after that, there it does not appear and this error message appears in the Xcode console:

```
2017-08-03 14:44:41.371739-0400 shopkids[636:85394] *** Assertion failure in -[NSLayoutConstraint _setMultiplier:], /BuildRoot/Library/Caches/com.apple.xbs/Sources/Foundation/Foundation-1350/Layout.subproj/NSLayoutConstraint.m:206
2017-08-03 14:44:41.371912-0400 shopkids[636:85394] [APPBOY] Exception NSInternalInconsistencyException caught in Appboy method viewWillAppear:The graphic image has width 0.000000 and height 0.000000 and image URI https://appboy-images.com/image_uploads/files/598369c55c5a68720128985d/cropped.png?1501784517 for campaign dGVzdF90ZXN0: Multiplier is not finite!  That's illegal.  multiplier:nan
```

## Describe your environment.

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | iOS
 Platform Version        | iOS 10.3.3
 SDK Version             | 2.31.0 
Integration Method      | cocoapods
 Xcode Version           | e.g. Xcode 8.3.3
 Repro rate              | all the time (100%)
 
## What did you do?

Attempt to trigger this campaign: https://dashboard.appboy.com/engagement/campaigns/in_app_message_140/59566eec56ec31088707bfa9?page=0

## What did you expect to happen?

The in-app message would appear

## What happened instead?

The app received the message to open the campaign, but it got the above `NSInternalInconsistencyException` and didn't display the in-app message.

## Steps to reproduce

1. Preview the above in-app campaign.
2. The campaign will appear ONE time for the user, but never again after that unless the app is reinstalled. Even if the app is closed and re-opened from the test push, it will not appear. 

## Code Snippet

This error seems to be related to the problem:

```
2017-08-03 14:44:41.371739-0400 shopkids[636:85394] *** Assertion failure in -[NSLayoutConstraint _setMultiplier:], /BuildRoot/Library/Caches/com.apple.xbs/Sources/Foundation/Foundation-1350/Layout.subproj/NSLayoutConstraint.m:206
2017-08-03 14:44:41.371912-0400 shopkids[636:85394] [APPBOY] Exception NSInternalInconsistencyException caught in Appboy method viewWillAppear:The graphic image has width 0.000000 and height 0.000000 and image URI https://appboy-images.com/image_uploads/files/598369c55c5a68720128985d/cropped.png?1501784517 for campaign dGVzdF90ZXN0: Multiplier is not finite!  That's illegal.  multiplier:nan
```

## Are you doing any feature customizations that may relate to the issue? Can you share the code snippet?

No

## Project that demonstrates the issue

https://itunes.apple.com/us/app/shopkids-clothes-and-gifts-for-tots-tykes/id1248508890?mt=8

Note: I have emailed support@appboy.com about this as well, but based on the above error message I think this may be an issue with the iOS SDK. Also this is very time sensitive for us because we are unable to properly launch a promotional campaign because of this.

**Comment by tiffanylu** [comment]
Hi @marcgug,

Thanks for opening this issue! Our support team has received your ticket and we are collaborating with them on addressing the issue via our main support channel.

Best,
Appboy iOS Team

**Comment by fmkpro1984** [comment]
Hi @tiffanylu,

We're also experiencing this issue, not only on the Modal Message-Type but also on Full-Screen Message-Type. Any update on this issue would be very much appreciated.

Thank you!

Regards,
Fritz

**Comment by tiffanylu** [comment]
Hi @fmkpro1984,

Could you let us know what iOS SDK version you're on and how you integrated the iOS SDK? 

Thanks!

**Comment by tiffanylu** [comment]
It would also be helpful to get more information about your `SDWebImage` integration and version number, as well as any customized integration code. If you want to limit the amount of information shared on GitHub, feel free to open a support ticket by emailing support@appboy.com and let us know so that our team can follow up from there.

Thanks @fmkpro1984!

**Comment by chuganzy** [comment]
Hi, @tiffanylu. I'm an iOS engineer working with @fmkpro1984 on the same project. Actually he is not an engineer so I'll answer your questions instead.

Appboy SDK version: 2.29.1
SDWebImage version: 4.0.0

We do not use SDWebImage in our app side and only Appboy SDK depends on it.

**Comment by tiffanylu** [comment]
Thanks @chuganzy! Our team has identified a potential issue and are currently working on a fix. We'll update this issue with more details once we have a clearer idea of our release timeline.

Best,
Tiff

**Comment by Wenzhi** [comment]
Hey @chuganzy,

Can you share more detail about the issue for full in-app message type, like stack trace, steps to reproduce or device logs? We are working on the fix right now but can only reroduce the issue on modal type. It would be great if you can help us reproduce the issue on full type as well.

Thanks,
Wenzhi

**Comment by Wenzhi** [comment]
Hey @chuganzy, 

Please see the above comment and let us know more information about your crash. It would be great if you can help us confirm the fix will also solve your issue.

Thanks,
Wenzhi

**Comment by fmkpro1984** [comment]
@Wenzhi Hi Wenzhi! Apologies, we can't reproduce the full screen in-app message yet, maybe we mistook it with the modal popup. Is the issue related to modal solved yet with the latest sdk? Is there any timing we could expect ?

**Comment by fmkpro1984** [comment]
@Wenzhi Any update on the bug above, about the fix date ? Thank you!

**Comment by Wenzhi** [comment]
Hey @fmkpro1984, 

The bug fix will be in our next release, which should be out around the end of this month. Does the timeline work for you?

Thanks,
Wenzhi

**Comment by fmkpro1984** [comment]
@Wenzhi (CC: @chuganzy )
Thank you!!! Very much appreciated, looking forward to it!

**Comment by tiffanylu** [comment]
Hi @marcgug, @fmkpro1984, and @chuganzy,

We just released the fix for this issue with iOS SDK version 3.1.1. Please feel free to re-open this issue if the fix doesn't work for you. Thanks again for reporting, and shoutout in particular for @marcgug for all your help with information gathering and debugging!

Best,
Tiff
